tasks/src/main/scala/dagr/tasks/gatk/IndelRealigner.scala

```diff
@@ -0,0 +1,109 @@
+import dagr.core.tasksystem.{Pipeline, SimpleInJvmTask}
+import dagr.sopt.{arg, clp}
+import dagr.tasks.DagrDef._
+import dagr.tasks.misc.DeleteFiles
+
+import scala.collection.mutable.ListBuffer
+
+@clp(
+  description="Run GATK indel cleaning on one or more BAMs together. Output BAMs are created in the output directory " +
+              "with name = basename(in) + .cleaned. + suffix(in).",
+  hidden=true
+)
+class IndelRealignment(val bams: Map[PathToBam,PathToBam],
+                       ref: PathToFasta,
+                       val known: Seq[PathToVcf],
+                       intervals: Option[PathToIntervals]
+                      ) extends Pipeline {
+
+  /** CLP constructor for testing and manual use, since there's no easy way to provide a map on the CLI. */
+  def this( @arg(flag="i", doc="One or more input BAM files.")       in : Seq[PathToBam],
+            @arg(flag="o", doc="Output directory.")                  out: DirPath,
+            @arg(flag="r", doc="Path to the reference fasta.")       ref: PathToFasta,
+            @arg(flag="k", doc="Zero or more VCFs of known indels.") known: Seq[PathToVcf] = Nil,
+            @arg(flag="l", doc="Optional regions to run over")       intervals: Option[PathToIntervals] = None) = {
+    this(Map[PathToBam,PathToBam] ( in map {b => b -> out.resolve(PathUtil.basename(b) + ".cleaned" + PathUtil.extensionOf(b).getOrElse(""))}:_*),
+         ref=ref,
+         known=known,
+         intervals=intervals)
+    Files.createDirectories(out)
+  }
+
+  /**
+    * Override this and build your pipeline there. Use the 'add' method
+    * to add tasks and make sure to define dependents between them.
+    */
+  override def build(): Unit = {
+    val ioMap: FilePath        = Files.createTempFile("indel_cleaning.", ".map")
+    val indelCleaningIntervals = Files.createTempFile("indel_cleaning.", ".intervals")
+    val targetCreator = new RealignerTargetCreator(in=bams.keys.toSeq, out=indelCleaningIntervals, known=known, ref=ref, intervals=intervals)
+    val makeIoMap = SimpleInJvmTask("MakeIndelRealignerMap", {
+      val lines = bams.map(pair => pair._1.getFileName.toString + "\t" + pair._2.toString).toSeq
+      Io.writeLines(ioMap, lines)
+    })
+    val indelRealigner = new IndelRealigner(in=bams.keys.toSeq, inOutMap=ioMap, ref=ref, known=known, targetIntervals=indelCleaningIntervals)
+    val cleanup = new DeleteFiles(ioMap, indelCleaningIntervals).withName("DeleteIndelTmpFiles")
+
+    root ==> (targetCreator :: makeIoMap) ==> indelRealigner ==> cleanup
+  }
+}
+
+/** Runs the realigner target creator step of the process. */
+class RealignerTargetCreator(val in : Seq[PathToBam],
+                             val out: FilePath,
+                             val known: Seq[PathToVcf],
+                             ref: PathToFasta,
+                             intervals: Option[PathToIntervals])
+  extends GatkTask(walker="RealignerTargetCreator", ref=ref, intervals=intervals) {
```

Comment:
I think this would just add another level of indirection but be no safer - what's to stop you typo'ing the enum value name?